packages/itwinui-react/src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -186,14 +188,12 @@ const OverflowGroup = React.forwardRef((props, forwardedRef) => {
     [childrenProp],
   );
 
-  const [overflowRef, visibleCount] = useOverflow(
-    items,
-    !overflowButton,
-    orientation,
-  );
-
-  return (
-    <BaseGroup
+  return overflowButton != null ? (
+    <OverflowContainer
+      as={BaseGroup}
+      items={items}
+      overflowOrientation={orientation}
+      overflowLocation={overflowPlacement}
       orientation={orientation}
```

Comment:
Looks like you have both `orientation` and `overflowOrientation`. Is this a mistake?

Reply:
It's fine for now, but I'm starting to believe we need to rethink our polymorphic `as` prop. I'm reminded of the [other recent issue](https://github.com/iTwin/iTwinUI/pull/2135#discussion_r1672595007) we had in `VisuallyHidden`. Let's make a note about this.

Adding an [`asChild` prop](https://www.radix-ui.com/primitives/docs/guides/composition#composing-multiple-primitives) could be one alternative to explore in the future.